Assets.Scripts.Core.Buriko/BurikoMemory.cs

```diff
@@ -48,6 +48,7 @@ public BurikoMemory()
 			variableReference.Add("GFlag_GameClear", 1);
 			variableReference.Add("GQsaveNum", 2);
 			variableReference.Add("GOnikakushiDay", 3);
+			variableReference.Add("GHighestChapter", 3); // For consistency between chapters
```

Comment:
@tellowkrinkle That's clever, but the problem is each arc has a different flag and number slot.   Here's wata for example.

https://github.com/07th-mod/higurashi-assembly/blob/15d1f9d8ad841edcad527db9516ecc28a333f444/Assets.Scripts.Core.Buriko/BurikoMemory.cs#L52

Reply:
I did realize that and will be very careful to update the number when merging

I originally added it to reduce errors like this one (from `hima-mg`) where they referenced variables from previous games instead of the current one, breaking the ChapterJump unlocks (I checked the game scripts and they do still use `GHimatsubushiDay`)

https://github.com/07th-mod/higurashi-assembly/blob/1a81ef4a52dbd9dda08c8225a70ce91659f8f1ba/Assets.Scripts.UI.ChapterJump/ChapterJumpButton.cs#L68